<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0" stdsetdef="1">
  <author></author>
  <comment></comment>
  <exportmacro></exportmacro>
  <class>HistoryForm</class>
  <widget class="QDialog" name="HistoryForm">
    <property name="geometry">
      <rect>
        <x>0</x>
        <y>0</y>
        <width>872</width>
        <height>647</height>
      </rect>
    </property>
    <property name="windowTitle">
      <string>Twinkle - Call History</string>
    </property>
    <layout class="QGridLayout">
      <item row="0" column="0">
        <widget class="Q3ListView" name="historyListView">
          <property name="allColumnsShowFocus">
            <bool>true</bool>
          </property>
          <property name="showSortIndicator">
            <bool>true</bool>
          </property>
          <property name="resizeMode">
            <enum>Q3ListView::LastColumn</enum>
          </property>
          <column>
            <property name="text">
              <string>Time</string>
            </property>
            <property name="clickable">
              <bool>true</bool>
            </property>
            <property name="resizable">
              <bool>true</bool>
            </property>
          </column>
          <column>
            <property name="text">
              <string>In/Out</string>
            </property>
            <property name="clickable">
              <bool>true</bool>
            </property>
            <property name="resizable">
              <bool>true</bool>
            </property>
          </column>
          <column>
            <property name="text">
              <string>From/To</string>
            </property>
            <property name="clickable">
              <bool>true</bool>
            </property>
            <property name="resizable">
              <bool>true</bool>
            </property>
          </column>
          <column>
            <property name="text">
              <string>Subject</string>
            </property>
            <property name="clickable">
              <bool>true</bool>
            </property>
            <property name="resizable">
              <bool>true</bool>
            </property>
          </column>
          <column>
            <property name="text">
              <string>Status</string>
            </property>
            <property name="clickable">
              <bool>true</bool>
            </property>
            <property name="resizable">
              <bool>true</bool>
            </property>
          </column>
        </widget>
      </item>
      <item row="1" column="0">
        <layout class="QHBoxLayout">
          <item>
            <widget class="Q3GroupBox" name="cdrGroupBox">
              <property name="title">
                <string>Call details</string>
              </property>
              <layout class="QGridLayout">
                <item row="0" column="0">
                  <widget class="Q3TextEdit" name="cdrTextEdit">
                    <property name="textFormat">
                      <enum>Qt::RichText</enum>
                    </property>
                    <property name="wordWrap">
                      <enum>Q3TextEdit::NoWrap</enum>
                    </property>
                    <property name="readOnly">
                      <bool>true</bool>
                    </property>
                    <property name="autoFormatting">
                      <set>Q3TextEdit::AutoAll</set>
                    </property>
                    <property name="whatsThis" stdset="0">
                      <string>Details of the selected call record.</string>
                    </property>
                  </widget>
                </item>
              </layout>
            </widget>
          </item>
          <item>
            <widget class="Q3GroupBox" name="viewGroupBox">
              <property name="title">
                <string>View</string>
              </property>
              <layout class="QVBoxLayout">
                <item>
                  <widget class="QCheckBox" name="inCheckBox">
                    <property name="text">
                      <string>&amp;Incoming calls</string>
                    </property>
                    <property name="shortcut">
                      <string>Alt+I</string>
                    </property>
                    <property name="whatsThis" stdset="0">
                      <string>Check this option to show incoming calls.</string>
                    </property>
                  </widget>
                </item>
                <item>
                  <widget class="QCheckBox" name="outCheckBox">
                    <property name="text">
                      <string>&amp;Outgoing calls</string>
                    </property>
                    <property name="shortcut">
                      <string>Alt+O</string>
                    </property>
                    <property name="whatsThis" stdset="0">
                      <string>Check this option to show outgoing calls.</string>
                    </property>
                  </widget>
                </item>
                <item>
                  <widget class="QCheckBox" name="successCheckBox">
                    <property name="text">
                      <string>&amp;Answered calls</string>
                    </property>
                    <property name="shortcut">
                      <string>Alt+A</string>
                    </property>
                    <property name="whatsThis" stdset="0">
                      <string>Check this option to show answered calls.</string>
                    </property>
                  </widget>
                </item>
                <item>
                  <widget class="QCheckBox" name="missedCheckBox">
                    <property name="text">
                      <string>&amp;Missed calls</string>
                    </property>
                    <property name="shortcut">
                      <string>Alt+M</string>
                    </property>
                    <property name="whatsThis" stdset="0">
                      <string>Check this option to show missed calls.</string>
                    </property>
                  </widget>
                </item>
                <item>
                  <widget class="QCheckBox" name="profileCheckBox">
                    <property name="text">
                      <string>Current &amp;user profiles only</string>
                    </property>
                    <property name="shortcut">
                      <string>Alt+U</string>
                    </property>
                    <property name="whatsThis" stdset="0">
                      <string>Check this option to show only calls associated with this user profile.</string>
                    </property>
                  </widget>
                </item>
              </layout>
            </widget>
          </item>
        </layout>
      </item>
      <item row="3" column="0">
        <layout class="QHBoxLayout">
          <item>
            <widget class="QPushButton" name="clearPushButton">
              <property name="text">
                <string>C&amp;lear</string>
              </property>
              <property name="shortcut">
                <string>Alt+L</string>
              </property>
              <property name="whatsThis" stdset="0">
                <string>&lt;p&gt;Clear the complete call history.&lt;/p&gt;
&lt;p&gt;&lt;b&gt;Note:&lt;/b&gt; this will clear &lt;b&gt;all&lt;/b&gt; records, also records not shown depending on the checked view options.&lt;/p&gt;</string>
              </property>
            </widget>
          </item>
          <item>
            <spacer name="spacer28">
              <property name="sizeHint">
                <size>
                  <width>540</width>
                  <height>20</height>
                </size>
              </property>
              <property name="sizeType">
                <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="orientation">
                <enum>Qt::Horizontal</enum>
              </property>
            </spacer>
          </item>
          <item>
            <widget class="QPushButton" name="closePushButton">
              <property name="text">
                <string>Clo&amp;se</string>
              </property>
              <property name="shortcut">
                <string>Alt+S</string>
              </property>
              <property name="default">
                <bool>false</bool>
              </property>
              <property name="whatsThis" stdset="0">
                <string>Close this window.</string>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QPushButton" name="callPushButton">
              <property name="text">
                <string>&amp;Call</string>
              </property>
              <property name="shortcut">
                <string>Alt+C</string>
              </property>
              <property name="default">
                <bool>true</bool>
              </property>
              <property name="whatsThis" stdset="0">
                <string>Call selected address.</string>
              </property>
            </widget>
          </item>
        </layout>
      </item>
      <item row="2" column="0">
        <layout class="QHBoxLayout">
          <item>
            <widget class="QLabel" name="numberlCallsTtextLabel">
              <property name="text">
                <string>Number of calls:</string>
              </property>
              <property name="wordWrap">
                <bool>false</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QLabel" name="numberCallsValueTextLabel">
              <property name="text">
                <string>###</string>
              </property>
              <property name="wordWrap">
                <bool>false</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QLabel" name="totalDurationTextLabel">
              <property name="text">
                <string>Total call duration:</string>
              </property>
              <property name="wordWrap">
                <bool>false</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QLabel" name="totalDurationValueTextLabel">
              <property name="text">
                <string>###</string>
              </property>
              <property name="wordWrap">
                <bool>false</bool>
              </property>
            </widget>
          </item>
          <item>
            <spacer name="spacer11">
              <property name="sizeHint">
                <size>
                  <width>460</width>
                  <height>20</height>
                </size>
              </property>
              <property name="sizeType">
                <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="orientation">
                <enum>Qt::Horizontal</enum>
              </property>
            </spacer>
          </item>
        </layout>
      </item>
    </layout>
  </widget>
  <layoutdefault spacing="6" margin="11"/>
  <pixmapfunction></pixmapfunction>
  <tabstops>
    <tabstop>historyListView</tabstop>
    <tabstop>cdrTextEdit</tabstop>
    <tabstop>inCheckBox</tabstop>
    <tabstop>outCheckBox</tabstop>
    <tabstop>successCheckBox</tabstop>
    <tabstop>missedCheckBox</tabstop>
    <tabstop>profileCheckBox</tabstop>
    <tabstop>clearPushButton</tabstop>
    <tabstop>closePushButton</tabstop>
  </tabstops>
  <includes>
    <include location="local">user.h</include>
    <include location="local">phone.h</include>
    <include location="local">q3popupmenu.h</include>
  </includes>
  <connections>
    <connection>
      <sender>closePushButton</sender>
      <signal>clicked()</signal>
      <receiver>HistoryForm</receiver>
      <slot>close()</slot>
    </connection>
    <connection>
      <sender>historyListView</sender>
      <signal>currentChanged(Q3ListViewItem*)</signal>
      <receiver>HistoryForm</receiver>
      <slot>showCallDetails(Q3ListViewItem*)</slot>
    </connection>
    <connection>
      <sender>inCheckBox</sender>
      <signal>toggled(bool)</signal>
      <receiver>HistoryForm</receiver>
      <slot>loadHistory()</slot>
    </connection>
    <connection>
      <sender>missedCheckBox</sender>
      <signal>toggled(bool)</signal>
      <receiver>HistoryForm</receiver>
      <slot>loadHistory()</slot>
    </connection>
    <connection>
      <sender>outCheckBox</sender>
      <signal>toggled(bool)</signal>
      <receiver>HistoryForm</receiver>
      <slot>loadHistory()</slot>
    </connection>
    <connection>
      <sender>profileCheckBox</sender>
      <signal>toggled(bool)</signal>
      <receiver>HistoryForm</receiver>
      <slot>loadHistory()</slot>
    </connection>
    <connection>
      <sender>successCheckBox</sender>
      <signal>toggled(bool)</signal>
      <receiver>HistoryForm</receiver>
      <slot>loadHistory()</slot>
    </connection>
    <connection>
      <sender>historyListView</sender>
      <signal>rightButtonPressed(Q3ListViewItem*,QPoint,int)</signal>
      <receiver>HistoryForm</receiver>
      <slot>popupMenu(Q3ListViewItem*,QPoint)</slot>
    </connection>
    <connection>
      <sender>historyListView</sender>
      <signal>doubleClicked(Q3ListViewItem*,QPoint,int)</signal>
      <receiver>HistoryForm</receiver>
      <slot>call(Q3ListViewItem*)</slot>
    </connection>
    <connection>
      <sender>clearPushButton</sender>
      <signal>clicked()</signal>
      <receiver>HistoryForm</receiver>
      <slot>clearHistory()</slot>
    </connection>
    <connection>
      <sender>callPushButton</sender>
      <signal>clicked()</signal>
      <receiver>HistoryForm</receiver>
      <slot>call()</slot>
    </connection>
  </connections>
</ui>
